src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,31 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod)), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * Prevents a lot of redundant calls to ContextFactory::createForNamespace.
+     *
+     * @param \ReflectionClass|\ReflectionMethod $reflector
+     */
+    private function createFromReflector($reflector): Context
+    {
+        if ($reflector instanceOf \ReflectionMethod)) {
+            $reflector = $reflector->getDeclaringClass();
+        }
+
+        $cacheKey = $reflector->getFileName();
```

Comment:
Hum, reading the source of the original `createFromReflector`, I get why the namespace was par of it.
It should be added back, sorry :)
Because the filename can be empty for internal/extension classes.

Reply:
done